llmstudio/llm/__init__.py

```diff
@@ -62,6 +76,38 @@ async def async_chat(self, input: str, is_stream=False, **kwargs):
         else:
             return await self.async_non_stream(input)
 
+    async def chat_coroutine(self, input: Union[str, List[Dict[str, str]]], semaphore: asyncio.Semaphore, max_retries: int = 5):
+        async with semaphore:
+            for i in range(max_retries):
+                try:
+                    # If the pause flag is set, wait for a while
+                    if self.pause:
+                        await asyncio.sleep(60)  # Wait for 60 seconds
```

Comment:
It doesn't make sense to add the pause inside the retry loop as we're already doing exponential backoff, this delay should only be added *before* a new co-routine starts looping so that we do not introduce new co-routines while we're already overwhelming the server